File: Programmers/오픈채팅방.py
def solution(record):
    answer = []
    
    # 아래 loop 에서는 최종 아이디만 기록
    dict = {}
    for i in record:
        current = i.split(' ')
        if current[0] != 'Leave' :
            dict[current[1]] = current[2]
        
    # 아래 loop 에서는 결과 입력
    for j in record:
        current = j.split(' ')
        if current[0] == 'Enter':
            answer.append(dict[current[1]] + "님이 들어왔습니다.")
        elif current[0] == 'Leave':
            answer.append(dict[current[1]] + "님이 나갔습니다.")
    
    return answer

File: JavaScript/오픈채팅방.js
function solution(record) {
    var answer = [];

    let ids = {};
    for (let i = 0; i < record.length; i++) {
        let rec = record[i].split(' ');
        if (rec[0] === 'Enter' || rec[0] === 'Change') ids[rec[1]] = rec[2];
    }
    
    for (let j = 0; j < record.length; j++) {
        let rec = record[j].split(' ');
        if (rec[0] === 'Enter') answer.push(ids[rec[1]] + '님이 들어왔습니다.');
        else if (rec[0] === 'Leave') answer.push(ids[rec[1]] + '님이 나갔습니다.');       
    }

    return answer;
}

console.log(solution(["Enter uid1234 Muzi", "Enter uid4567 Prodo","Leave uid1234","Enter uid1234 Prodo","Change uid4567 Ryan"]));
